code/game/turfs/simulated.dm

```diff
@@ -18,26 +19,64 @@
 
 	roof_type = /turf/simulated/floor/airless/ceiling
 
-/turf/simulated/proc/wet_floor(var/wet_val = 1)
-	if(wet_val < wet)
-		return
-
-	if(!wet)
-		wet = wet_val
+/turf/simulated/proc/wet_floor(var/type = WET_TYPE_WATER, var/amount = 1)

```

Comment:
`type` is a `/datum` var; masking it with a param could be a source of confusion.